michbl/assignment3/main.cpp

```diff
@@ -1,42 +1,128 @@
 #include <iostream>
+#include <set>
 #include "Dictionary.hpp"
 #include "Grid.hpp"
 #include "codeu_test_lib.h"
 
+enum Direction {
+    LEFT, RIGHT, UP, DOWN, TOP_LEFT, TOP_RIGHT, BOTTOM_LEFT, BOTTOM_RIGHT
+};
+
+// function declarations
+void getNextIdxs(const int row, const int col, const Direction dir, int &nextRow, int &nextCol);
+
+set<string> wordSearch(const Dictionary &dict, const Grid &grid);
+
+void wordSearchBT(const Dictionary &dict, const Grid &grid, int row, int col, set<string> &res, string word);
```

Comment:
No need to separate declarations and definitions for these assignments. You can mostly avoid the need for declarations by writing the functions in the order you use them.
(Unless required by the compiler for some reason, e.g. function A which calls functions B which calls function A)